core/forms.py

```diff
@@ -0,0 +1,70 @@
+from django import forms
+from django.contrib.auth import authenticate
+from django.contrib.auth.forms import AuthenticationForm
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext, gettext_lazy as _
+
+##################################################################################
+# Defines general-purpose forms.
+# @author E.M.A. Arts
+# @since 15 AUG 2019
+##################################################################################
+
+
+# LoginForm that changes the default AuthenticationForm
+# It provides a different error message when passing invalid login credentials,
+# and allows Inactive users to login
+class LoginForm(AuthenticationForm):
+
+    def clean(self):
+        # Obtain username and password
+        username = self.cleaned_data.get('username')
+        password = self.cleaned_data.get('password')
+
+        # Check if both are provided
+        if username is not None and password:
```

Comment:
What should this if statement do? It looks like it will only be True if password is True?
Should it be if (username is not None) and (password is not None)?

Reply:
This statement utilises python's type conversion. E.g. Expressions such as `False`, `""` (the empty string), `None`, or `[]` (the empty list) will all result in `False`.
These should not be valid for a username either, so it should actually be `if username and password:`, good catch!